cmd/serveMCP.go

```diff
@@ -0,0 +1,133 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"regexp"
+	"strconv"
+
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mark3labs/mcp-go/server"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var serveMcpCmd = &cobra.Command{
+	Use:   "serve-mcp",
+	Short: "Starts the poutine MCP server",
+	Long: `Starts the poutine MCP server.
+Example to start the MCP server: poutine serve-mcp --token "$GH_TOKEN"`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		Token = viper.GetString("token")
+		ctx := cmd.Context()
+		s := server.NewMCPServer("poutine", Version)
+		analyzer, err := GetAnalyzer(ctx, "")
```

Comment:
@fproulx-boostsecurity yeah initializing poutine this way won't work properly because it looks for var already assigned in the root cmd to set scm params and stuff, to make it not be hardcoded to just Github we will have to bring all those root params here and have them exposed in the mcp server